src/transformers/models/llava_next/modeling_llava_next.py

```diff
@@ -64,11 +65,51 @@ def get_anyres_image_grid_shape(image_size, grid_pinpoints, patch_size):
     """
     if not isinstance(grid_pinpoints, list):
         raise ValueError("grid_pinpoints should be a list of tuples or lists")
+    
+    # ! VERY IMPORTANT if image_size is tensor, must convert to into tuple, otherwise it will cause wrong calculate
+    if not isinstance(image_size, (list, tuple)):
+        assert isinstance(image_size, (torch.Tensor, np.ndarray)), f'image_size invalid type: {type(image_size)} | {image_size}'
```

Comment:
we don't use assertions in the code base but rather `if .... else raise ValueError`